GitCommands/Git/Executable.cs

```diff
@@ -177,15 +177,18 @@ public StreamReader StandardError
             /// <inheritdoc />
             public void Dispose()
             {
-                if (Interlocked.CompareExchange(ref _disposed, 1, 0) != 0)
+                lock (_syncRoot)
                 {
-                    return;
+                    if (_disposed == 0)
+                    {
+                        _disposed = 1;
+                        return;
+                    }
                 }
 
-                _process.Exited -= OnProcessExit;
```

Comment:
Ditto. I know it looks cleaner but I spent a lot of time tweaking the order of things here. Can't remember if this particular case is important, but if you only changed it to look nicer then consider reverting this reordering.